tests.py

```diff
@@ -774,48 +714,112 @@ def test_fully_coherent_MCMC_skyprior(self):
             Delta=self.Delta,
             tref=self.tref,
             duration=self.duration,
-            Band=4,
+            Band=self.Band,
         )
         Writer.make_data()
 
         predicted_FS = Writer.predict_fstat()
 
-        theta = {
-            "F0": {"type": "norm", "loc": self.F0, "scale": np.abs(1e-10 * self.F0)},
-            "F1": {"type": "norm", "loc": self.F1, "scale": np.abs(1e-10 * self.F1)},
-            "F2": self.F2,
-            "Alpha": {
-                "type": "unif",
-                "lower": self.Alpha - 0.01,
-                "upper": self.Alpha + 0.01,
+        # use a single test case with loop over multiple prior choices
+        # this could be much more elegantly done with @pytest.mark.parametrize
+        # but that cannot be mixed with unittest classes
+        thetas = {
+            "uniformF0-uniformF1-fixedSky": {
+                "F0": {
+                    "type": "unif",
+                    "lower": self.F0 - 1e-6,
+                    "upper": self.F0 + 1e-6,
+                },
+                "F1": {
+                    "type": "unif",
+                    "lower": self.F1 - 1e-10,
+                    "upper": self.F1 + 1e-10,
+                },
+                "F2": self.F2,
+                "Alpha": self.Alpha,
+                "Delta": self.Delta,
+            },
+            "log10uniformF0-uniformF1-fixedSky": {
+                "F0": {
+                    "type": "log10unif",
+                    "log10lower": np.log10(self.F0 - 1e-6),
+                    "log10upper": np.log10(self.F0 + 1e-6),
+                },
+                "F1": {
+                    "type": "unif",
+                    "lower": self.F1 - 1e-10,
+                    "upper": self.F1 + 1e-10,
+                },
+                "F2": self.F2,
+                "Alpha": self.Alpha,
+                "Delta": self.Delta,
+            },
+            "normF0-normF1-fixedSky": {
+                "F0": {"type": "norm", "loc": self.F0, "scale": 1e-6},
+                "F1": {"type": "norm", "loc": self.F1, "scale": 1e-10},
+                "F2": self.F2,
+                "Alpha": self.Alpha,
+                "Delta": self.Delta,
+            },
+            "lognormF0-halfnormF1-fixedSky": {
```

Comment:
I'll work this out to see if it's right

Reply:
Ok, so `loc` is used to place the distribution at the right place (whatever that means, since the mode of a lognormal is not located at `exp(mu)`, but `exp(mu - sigma^2)` ), and you need to specify it as a logarithm because you will be using `numpy`'s lognorm in the end (you would use the actual value if it was scipy).

Reply:
Correct.